experiments/AtmosGCM/moist-baroclinic-wave.jl

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env julia --project
+using ClimateMachine
+ClimateMachine.init()
+using ClimateMachine.Atmos
+using ClimateMachine.ConfigTypes
+using ClimateMachine.Diagnostics
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.SystemSolvers: ManyColumnLU
+using ClimateMachine.Mesh.Filters
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.Thermodynamics: total_energy, air_density
+using ClimateMachine.VariableTemplates
+
+using LinearAlgebra
+using StaticArrays
+using Test
+
+using CLIMAParameters
+using CLIMAParameters.Planet: R_d, day, grav, planet_radius, Omega
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+function init_moist_baroclinic_wave!(bl, state, aux, coords, t)
+    FT = eltype(state)
+
+    φ = latitude(bl, aux)
+    λ = longitude(bl, aux)
+    z = altitude(bl, aux)
+
+    # parameters 
+    _grav::FT = grav(bl.param_set)
+    _R_d::FT = R_d(bl.param_set)
+    _Ω::FT = Omega(bl.param_set)
+    _a::FT = planet_radius(bl.param_set)
+    
+    T_E::FT = 310
+    T_P::FT = 240
+    T_0::FT = 0.5 * (T_E + T_P)
+    Γ::FT = 0.005
+    b::FT = 2
+    K::FT = 3
+    p_0::FT = 1e5
+    M_v::FT = 0.608
+    z_p::FT = 15e3
+    λ_p::FT = π/9
+    φ_p::FT = 2*π/9
+    R_p::FT = _a/10
+    u_p::FT = 1
+    q_0::FT = 0.018
+    φ_w::FT = 2*π/9
+    p_w::FT = 0.34e5
+    p_t::FT = 0.1e5
+    q_t::FT = 1e-12 
+
+    # convenience functions for temperature and pressure
+    I_T::FT = cos(φ)^K - K/(K+2) * cos(φ)^(K+2) 
+    τ_z_1::FT = exp(Γ*z/T_0)
+    τ_z_2::FT = 1 - 2*(z*_grav/b/_R_d/T_0)^2
+    τ_z_3::FT = exp(-(z*_grav/b/_R_d/T_0)^2)
+    τ_1::FT = 1/T_0 * τ_z_1 + (T_0-T_P)/T_0/T_P * τ_z_2 * τ_z_3 
+    τ_2::FT = 0.5 * (K+2) * (T_E-T_P)/T_E/T_P * τ_z_2 * τ_z_3 
+    τ_int_1::FT = 1/Γ * (τ_z_1-1) + z * (T_0-T_P)/T_0/T_P * τ_z_2 
+    τ_int_2::FT = 0.5 * (K+2) * (T_E-T_P)/T_E/T_P * z * τ_z_3
+
+    # temperature, pressure, specific humidity, density
+    T_v::FT = 1 / (τ_1 - τ_2 * I_T) 
+    p::FT = p_0 * exp(-_grav/_R_d * (τ_int_1 - τ_int_2 * I_T))
+    η::FT = p/p_0
+    #q::FT = q_0 * exp(-(φ/φ_w)^4) * exp(-((η-1)*p_0/p_w)^2)
```

Comment:
I suppose this is because you want to run the dry case first ?

Reply:
:)
correct